lighthouse/classes/event_properties/interfaces.py

```diff
@@ -0,0 +1,225 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Dict
+from lighthouse.classes.messages.warehouse_messages import WarehouseMessage  # type: ignore
+from contextlib import contextmanager
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ValidationError(Exception):
+    pass
+
+
+class RetrievalError(Exception):
+    pass
+
+
+class EventPropertyInterface(ABC):
+    """
+    This class defines the public interface offered by an event property.
+    Any method we need to have present in event property should be defined
+    inside this class.
+    """
+
+    @abstractmethod
+    def validate(self) -> bool:
+        """
+        Retuns a boolean (True or False) indicating if the params received for this
+        EventProperty are correct in order to retrieve the data. If there is an error
+        in validation it will add it to the errors list.
+        This methods is safe and should not raise any exception.
+
+        Arguments:
+            None
+
+        Returns:
+            bool - True/False depending on the result of the validation.
+
+        """
+        ...
+
+    @abstractmethod
+    def errors(self) -> List[str]:
```

Comment:
Is this a property? See https://docs.python.org/3/library/abc.html#abc.abstractproperty, you can use the `property` decorator in the abstract class now.

Reply:
Much better